granary/flickr.py

```diff
@@ -0,0 +1,371 @@
+from apiclient.errors import HttpError
+from apiclient.http import BatchHttpRequest
+from oauth_dropins.webutil import util
+from oauth_dropins import flickr_auth
+
+import appengine_config
+import datetime
+import functools
+import itertools
+import json
+import logging
+import requests
+import source
+import sys
+import mf2py
+import urllib2
+
+
+class Flickr(source.Source):
+
+  DOMAIN = 'flickr.com'
+  NAME = 'Flickr'
+
+  API_EXTRAS = ','.join(('date_upload', 'date_taken', 'views', 'media',
+                         'description' 'tags', 'machine_tags', 'geo'))
+
+  def __init__(self, access_token_key, access_token_secret, user_id=None):
+    """Constructor.
+
+    Args:
+      access_token_key: string, OAuth access token key
+      access_token_secret: string, OAuth access token secret
+      user_id: string, the logged in user's Flickr nsid (optional)
+    """
+    self.access_token_key = access_token_key
+    self.access_token_secret = access_token_secret
+    self.user_id = user_id
+
+  def call_api_method(self, method, params={}):
+    """Call a Flickr API method.
+    """
+    return flickr_auth.call_api_method(
+      method, params, self.access_token_key, self.access_token_secret)
+
+  def get_activities_response(self, user_id=None, group_id=None, app_id=None,
+                              activity_id=None, start_index=0, count=0,
+                              etag=None, min_id=None, cache=None,
+                              fetch_replies=False, fetch_likes=False,
+                              fetch_shares=False, fetch_events=False,
+                              search_query=None):
+    """Get Flickr actvities
+    """
+    if user_id is None:
+      user_id = 'me'
+    if group_id is None:
+      group_id = source.FRIENDS
+
+    params = {}
+    method = None
+    solo = False
```

Comment:
very minor: this is equivalent to `activity_id` (boolean conversion), right? maybe just use that directly?